### DISCUSSION TITLE
source install on pg11 and pg12 fails

### DISCUSSION THREAD

**Discussion by snps-anmolm** [apache/age#386] [Q&A]
I'm running postgresql 11.18, and trying to do source install for age. But getting below error

>>:/apacheage/age-PG12-v1.1.0-rc0 % make PG_CONFIG=.../postgres/bin/pg_config install
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -O2 -fPIC -I.//src/include -I.//src/include/parser -I. -I./ -I/../postgres/include/server -I/../postgres/include/internal  -D_GNU_SOURCE   -c -o src/backend/age.o src/backend/age.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -O2 -fPIC -I.//src/include -I.//src/include/parser -I. -I./ -I/../postgres/include/server -I/../postgres/include/internal  -D_GNU_SOURCE   -c -o src/backend/catalog/ag_catalog.o src/backend/catalog/ag_catalog.c

gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -O2 -fPIC -I.//src/include -I.//src/include/parser -I. -I./ -I/../postgres/include/server -I/../postgres/include/internal  -D_GNU_SOURCE   -c -o src/backend/catalog/ag_graph.o src/backend/catalog/ag_graph.c
src/backend/catalog/ag_graph.c: In function ‘insert_graph’:
src/backend/catalog/ag_graph.c:55:5: warning: implicit declaration of function ‘table_open’
src/backend/catalog/ag_graph.c:55:14: warning: assignment makes pointer from integer without a cast
src/backend/catalog/ag_graph.c:73:5: warning: implicit declaration of function ‘table_close’
src/backend/catalog/ag_graph.c: In function ‘delete_graph’:
src/backend/catalog/ag_graph.c:87:14: warning: assignment makes pointer from integer without a cast
src/backend/catalog/ag_graph.c: In function ‘update_graph_name’:
src/backend/catalog/ag_graph.c:121:14: warning: assignment makes pointer from integer without a cast
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -O2 -fPIC -I.//src/include -I.//src/include/parser -I. -I./ -I/../postgres/include/server -I/../postgres/include/internal  -D_GNU_SOURCE   -c -o src/backend/catalog/ag_label.o src/backend/catalog/ag_label.c
In file included from src/backend/catalog/ag_label.c:43:0:
.//src/include/executor/cypher_utils.h:24:26: fatal error: access/table.h: No such file or directory
compilation terminated.
make: *** [src/backend/catalog/ag_label.o] Error 1

**Comment by MuhammadTahaNaveed** [comment]
> :/apacheage/age-PG12-v1.1.0-rc0

Hello, actually the problem is that you are trying to install AGE that is for pg version > 12.0. For `postgresql 11.18`, downloading the source code from [here](https://github.com/apache/age/archive/refs/tags/v1.1.0-rc0.tar.gz) and then installing AGE will most probably resolve the issue.

If you still face issue with respect to installation, I recommend going through any of the below articles on Apache AGE installation.
- https://theundersurfers.netlify.app/age-installation/
- https://medium.com/@m.taha.naveed27/postgresql-and-apache-age-installation-from-source-linux-and-mac-9e4a1ea14fdf

**Comment by snps-anmolm** [comment]
Hello,

Thank you for quickly reply ! I even tried with PG12.10 (released 2022-02-07) and PG12.13 (released on 2022-11-07). Not same as above error, but in both cases I get below error

>:/path_to_age/apacheage/age-PG12-v1.1.0-rc0 $ make PG_CONFIG=/path_to_age/apacheage/postgres/bin/pg_config install
......
......
gcc -std=gnu99 -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -O2 -fPIC -I.//src/include -I.//src/include/parser -I. -I./ -I/path_to_age/apacheage/postgres/include/server -I/path_to_age/apacheage/postgres/include/internal  -D_GNU_SOURCE   -c -o src/backend/utils/adt/age_graphid_ds.o src/backend/utils/adt/age_graphid_ds.c
src/backend/utils/adt/age_graphid_ds.c:35:3: error: redefinition of typedef ‘GraphIdNode’
.//src/include/utils/age_graphid_ds.h:37:28: note: previous declaration of ‘GraphIdNode’ was here
src/backend/utils/adt/age_graphid_ds.c:43:3: error: redefinition of typedef ‘ListGraphId’
.//src/include/utils/age_graphid_ds.h:40:28: note: previous declaration of ‘ListGraphId’ was here
make: *** [src/backend/utils/adt/age_graphid_ds.o] Error 1

**Reply by snps-anmolm** [reply]
I tried with latest git repo and postgres 11.18 as mentioned in link shared by you - https://theundersurfers.netlify.app/age-installation/
but end up with same graphIdNode and ListGraphId issue. Is there issue with my gcc version here ?

**Reply by jrgemignani** [reply]
You need to install the PostgreSQL 11 version of AGE (current 11 is on the **master** branch) with PostgreSQL 11. 
For the PostgreSQL 12 version of AGE (current 12 is on the **PG12** branch)  with PostgreSQL 12.

You need to make sure that when you install one of them, you have the correct path for PG_CONFIG. It must point to the correct version of PostgreSQL for the version of AGE.

Additionally, I would do a make clean before a make install.
